File: ProblemSolving/Array/ReverseArray.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    void reverseString(vector<char> &s)
    {
        int len = s.size();
        for (int i = 0; i < s.size() / 2; i++)
        {
            swap(s[i], s[len - 1 - i]);
        }
    }
};

File: ProblemSolving/Array/ReverseArray.py
from ast import List
import math

# first solution


class Solution:
    def reverseString(self, s: List[str]) -> None:
        length = len(s)
        for i in range(math.floor(length/2)):
            s[i], s[length-i-1] = s[length-i-1], s[i]


# second solution
class Solution2:
    def reverseString(self, s: List[str]) -> None:
        s[:] = s[::-1]
